implementations/utils/vm_api.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import subprocess
+from typing import Optional, Sequence, TextIO
+from absl import logging
+
+
+class RunError(Exception):
+  pass
+
+
+class gpu_api:
```

Comment:
In my option, use BashOperator to implement `build()` will be simpler. What's your thoughts? @will-cromar

Reply:
Thanks, I didn't know we have `BashOperator`  to use. I can submit a separate PR to change to `BashOperator`.

Reply:
I think it will be clearer if we use BashOperator for that build(). A few examples here: https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/bash.html

Reply:
Let's also discuss it offline.